lib/bundler.rb

```diff
@@ -366,7 +366,11 @@ def requires_sudo?
 
         # if any directory is not writable, we need sudo
         files = [path, bin_dir] | Dir[path.join("build_info/*").to_s] | Dir[path.join("*").to_s]
-        sudo_needed = files.any? {|f| !File.writable?(f) }
+        unwritable_files = files.reject {|f| File.writable?(f) }
+        sudo_needed = !unwritable_files.empty?
+        if sudo_needed
+          Bundler.ui.warn "Following files may not be writable, so sudo is needed: #{unwritable_files.map(&:to_s).join(",")}"
```

Comment:
should this list be printed on multiple lines instead of being separated by `,`. I imagine this list could in circumstances be quiet big and having it all printed on a single line wouldn't look very nice to the user.